### PR TITLE
Dialogs: Move click handler from templates to JS

### PR DESCRIPTION
**Pull request by jdlrobson**
### Description
From now on simply adding a class .floaterShut to an element will make it close an open dialog.
.floaterShut--parent will request the parent element closes it.

Previously we did this via inline JavaScript which made it really hard to determine which pages should run this code. Having it in a central place also allows us in future to defer the loading of jquery ui off the critical path. This is the most simple first step!

### Testing
Visit http://localhost:8080/works/OL53924W/The_complete_works_of_Mark_Twain#addImage and click "change cover". The close icon should close the dialog

### CODE REVIEW

File: openlibrary/plugins/openlibrary/js/dialog.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Wires up dialog close buttons
+ */
+export default function dialogs() {
```
Comment (cdrini) [inline openlibrary/plugins/openlibrary/js/dialog.js]: `initDialogs` is more consistent; but I see you do that in a future commit :+1:

File: openlibrary/plugins/openlibrary/js/dialog.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    // This will close the dialog in the current page.
+    $('.floaterShut').attr('href', 'javascript:;').on('click', () => $.fn.colorbox.close());
```
Comment (cdrini) [inline openlibrary/plugins/openlibrary/js/dialog.js]: `floaterShut` is a weird name; Let's rename it to `dialog--close` and `dialog--close-parent`.

(There are currently 3 ways these things are referred to throughout the code: `floater`, `popup`, `dialog`, `colorbox`; we should consolidate to one.)

File: openlibrary/templates/covers/manage.html [hunk 3]
```diff
@@ -58,7 +58,7 @@
 
     <div class="formElement" style="margin:40px 0 0 30px;">
         <button type="submit" value="Save" class="largest">$_("Save")</button>
-        <a class="floaterClose close-popup" href="javascript:;" onclick="parent.\$.fn.colorbox.close();">$_("Cancel")</a>
+        <a class="floaterClose floaterShut--parent close-popup">$_("Cancel")</a>
```
Comment (cdrini) [inline openlibrary/templates/covers/manage.html]: `floaterClose` is unused (2refs); can you delete in a future PR? Or add to your WIP as a separate commit.

Comment (cdrini) [inline openlibrary/templates/covers/manage.html]: `close-popup` does the same thing (i.e. `$.fn.colorbox.close()`) (see automatic.js). Let's kill it in favor of your new methods. (Put in a separate commit, bitte)

File: openlibrary/plugins/openlibrary/js/dialog.js (hunk 2 above)
Comment (jdlrobson) [reply openlibrary/plugins/openlibrary/js/dialog.js]: Agreed. Do you mind if i do this in a follow up? It will be easier to  change all the instances then.

Do I need to worry about cached HTML? If so, I can do a combined selector to account for both e.g. ``` .floaterShut,.dialog--close```

File: openlibrary/templates/covers/manage.html (hunk 3 above)
Comment (jdlrobson) [reply openlibrary/templates/covers/manage.html]: I can remove this in a follow up when I introduce .dialog--close

Comment (jdlrobson) [reply openlibrary/templates/covers/manage.html]: Ditto above.

File: openlibrary/plugins/openlibrary/js/dialog.js (hunk 1 above)
Comment (jdlrobson) [reply openlibrary/plugins/openlibrary/js/dialog.js]: I can do that as part of this commit. I think this got lost in my rebasing.

**Comment by jdlrobson** [comment]
Are you okay for me to do the remaining changes as follow ups? I should have some time available tomorrow to do that as well as get #2423 ready.

File: openlibrary/plugins/openlibrary/js/dialog.js (hunk 2 above)
Comment (cdrini) [reply openlibrary/plugins/openlibrary/js/dialog.js]: I would rather have it error than have to do that :P Doing in a follow up sounds good :+1:

File: openlibrary/templates/covers/manage.html (hunk 3 above)
Comment (cdrini) [reply openlibrary/templates/covers/manage.html]: :+1:

**Comment by cdrini** [comment]
That seems good to me. Are you going to put them on top of your WIP PR?

File: openlibrary/plugins/openlibrary/js/dialog.js (hunk 2 above)
Comment (cdrini) [reply openlibrary/plugins/openlibrary/js/dialog.js]: Although we should look into automatically adding `?v=COMMIT_HASH` to the webpack dynamic imports; that would solve this issue.

**Comment by jdlrobson** [comment]
I've got a new patchset locally that ill push when this is merged.

**Comment by jdlrobson** [comment]
Okay, given it's not merged I'll temporary submit to my WIP branch. When it's merged I'll split that into reviewable separate pull requests.

**Comment by jdlrobson** [comment]
For completeness the follow ups are in https://github.com/internetarchive/openlibrary/pull/2423/commits/00a4db25f596c18c264c8902d4d62b70bca8722a

I can push a pull request tomorrow if this one is merged tonight/tomorrow

**Comment by cdrini** [comment]
Oh, whoops, missed the other commits on here. My apologies for the delay; I had weekend plans and wasn't able to finish this before heading out of town.

File: openlibrary/plugins/openlibrary/js/dialog.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
+import initTabs from './tabs';
+import { placeholder } from './jquery-ui';
+
+/**
+ * @param {string} selector of element that will open dialog
+ * @param {string} href of element that will be the contents of the dialog
+ */
+function setupDialog(selector, href) {
```
Comment (cdrini) [inline openlibrary/plugins/openlibrary/js/dialog.js]: Can we rename `href` to `dialogSelector`? `href` makes it seem like it could be an arbitrary link.

**Comment by jdlrobson** [comment]
The name href is used for consistency with the colorbox plugin which is
called href (http://www.jacklmoore.com/colorbox/)
 According to the docs

This can be used as an alternative anchor URL or to associate a URL for
non-anchor elements such as images or form buttons.
$("h1").colorbox({href:"welcome.html"});


I'd thus like us to use href too. It also means the code is less verbose as
the parameter can be used as the key name and value name. Would better docs
in this doc tag help?

**Comment by jdlrobson** [comment]
Sorry for confusion let's move discussion about clickHander

> Oh, whoops, missed the other commits on here. My apologies for the delay; I had weekend plans and wasn't able to finish this before heading out of town.

Sorry for confusion that wasn't intentional. We can continue the discussions on https://github.com/internetarchive/openlibrary/pull/2423- once this is merged.
